odl/trafos/backends/pywt_bindings.py

```diff
@@ -0,0 +1,523 @@
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Bindings to the ``PyWavelets`` backend for wavelet transforms."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+import numpy as np
+
+try:
+    import pywt
+    PYWAVELETS_AVAILABLE = True
+except ImportError:
+    PYWAVELETS_AVAILABLE = False
+
+
+__all__ = ('pywt_wbasis', 'pywt_coeff_shape_list',
+           'pywt_coeff_to_array', 'array_to_pywt_coeff',
+           'PAD_MODES_ODL2PYWT', 'PAD_MODES_ODL2PYWT',
+           'PYWT_SUPPORTED_PAD_MODES', 'PYWAVELETS_AVAILABLE')
+
+
+PAD_MODES_ODL2PYWT = {'constant': 'zpd',
+                      'periodic': 'ppd',
+                      'symmetric': 'sym',
+                      'order0': 'cpd',
+                      'order1': 'sp1',
+                      'pywt_periodic': 'per'}
+PAD_MODES_PYWT2ODL = {v: k for k, v in PAD_MODES_ODL2PYWT.items()}
+PYWT_SUPPORTED_PAD_MODES = PAD_MODES_ODL2PYWT.keys()
+
+
+def pywt_wbasis(wbasis):
+    """Convert ``wbasis`` a to `pywt.Wavelet` if it is not already one."""
+    if isinstance(wbasis, pywt.Wavelet):
+        return wbasis
+    else:
+        return pywt.Wavelet(wbasis)
+
+
+def pywt_max_level(*args, **kwargs):
+    """Return maximum number of scaling levels."""
+    if not PYWAVELETS_AVAILABLE:
+        raise NotImplementedError('PyWavelets not available')
+    return pywt.dwt_max_level(*args, **kwargs)
+
+
+def pywt_coeff_shape_list(shape, wbasis, nscales, pad_mode):
+    """Return a list of coefficient shapes in the specified transform.
+
+    Parameters
+    ----------
+    shape : tuple
+        Number of pixels/voxels in the image. Its length must be 1, 2 or 3.
+    wbasis : string or `pywt.Wavelet`
+        Specification of the wavelet to be used in the transform.
+        If a string is given, it is converted to a `pywt.Wavelet`.
+        For more information see the `wavelets
+        <http://www.pybytes.com/pywavelets/ref/wavelets.html>`_
+        page of the PyWavelets documentation.
+
+        Possible wavelet families are:
+
+        'Haar': Haar
+
+        'db': Daubechies
+
+        'sym' : Symlets
+
+        'coif': Coiflets
+
+        'bior': Biorthogonal
+
+        'rbio': Reverse biorthogonal
+
+        'dmey': Discrete FIR approximation of the Meyer wavelet
```

Comment:
I am strongly in favor of keeping this, mostly because "click the link" is not possible from a IDE.

With that said, this list is lacking some things for example `'db` is not a valid parameter, users need to give `dbN` for "some" `N`.